python-sdk/docs/guides/databricks.rst

```diff
@@ -0,0 +1,91 @@
+.. _databricks:
+
+======================
+Databricks Integration
+======================
+Introduction
+
+The Astro Python SDK now allows users to easily access and work with data stored in Databricks through the Delta API. The Astro Python SDK provides a simple, yet powerful interface for working with Spark and Delta tables in Databricks, allowing users to take advantage of the scalability and performance of Databricks while maintaining the flexibility of Airflow DAG authoring and all the benefits of Airflow scheduling.
+
+Installation
+============
+To use the Astro Python SDK with Databricks, complete the following steps:
+
+The first step is to install the databricks submodule of the astro-sdk-python pip library. This can be done by running the following command:
+
+.. code-block:: bash
+
+    pip install astro-sdk-python[databricks]
+
+The second step is to create a connection to your Databricks cluster.
+This requires creating a `personal access token <https://docs.databricks.com/dev-tools/api/latest/authentication.html>`_ in Databricks and creating a cluster with an http_endpoint. Once you have these, you can create a connection using the following syntax:
+
+
+.. code-block:: yaml
+
+    - conn_id: databricks_conn
+      conn_type: databricks
+      host: https://my-cluster.cloud.databricks.com/
+      password: my-token
+      extra:
+        http_path: /sql/1.0/warehouses/foobar
+
+Loading Data
+============
+Once you have installed the astro-sdk package and created a connection to your Databricks cluster, you can begin loading data into your Databricks cluster.
+
+To load data into your databricks cluster, you can user the same ``aql.load_file()`` function that works for all other databases.
+The only thing that is different with delta is that you now can use the ``delta_options`` parameter to specify delta specific parameters(such as ``copy_options`` and ``format_options`` for the ``COPY INTO`` command).
+Please note that when loading data into Delta using ``COPY INTO``, you must specify the filetype as Databricks does not automatically infer the data (this is not the case for autoloader).
+
+Currently, only local files can be loaded using the aql.load_file() function. Support for loading data from S3 and GCS will be added soon.
+
+To use the aql.load_file() function, you will need to specify the path to the file you want to load, the target Delta table, and any additional COPY INTO arguments using the databricks_options parameter. For example:
+
+.. code-block:: python
+
+    aql.load_file(
+        input_file=File("data.csv"),
+        output_table=Table(conn_id="my_databricks_conn"),
+        databricks_options={"copy_into_options": {"format_options": {"header": "true"}}},
```

Comment:
Currently, for loading other CSV to other databases, I believe the default option is for the file to have a header. Perhaps we should make this the default in this case as well - so users don't need to specify it all the time.

Reply:
That's a good call. It's annoying especially because having a header is default for autoloader but for some reason isn't for COPY INTO.